contracts/registry.go

```diff
@@ -19,11 +19,6 @@ func GetRegisteredAddress(vmRunner vm.EVMRunner, registryId common.Hash) (common
 	vmRunner.StopGasMetering()
 	defer vmRunner.StartGasMetering()
 
-	// TODO(mcortesi) remove registrypoxy deployed at genesis
-	if vmRunner.GetStateDB().GetCodeSize(params.RegistrySmartContractAddress) == 0 {
```

Comment:
Shouldn't we now be able to remove the GetStateDB method from the vm runner interface and all its implemenationns?

Reply:
https://github.com/celo-org/celo-blockchain/blob/c4ea1da3bcba2a1e8909efbc741e3d4c74c5d4f4/contracts/random/random.go#L72-L75

There is another call to `GetCodeSize()`

Reply:
@mcortesi Can you elaborate how this FIXME means?

Reply:
the second one is still required... but i'm 100% sure why (predates me).

My comment is because IMO the caller this function should do the `Finalise()` and not the function itself. The function as all other function, shoudl just make a contract call... if you want to finalse state after,, that's the caller's reponsability.

Reply:
The chunk could be safely delete, because two callers already called `stateDB.IntermediateRoot()`, which contains calling `Finalise()`.

https://github.com/celo-org/celo-blockchain/blob/c4ea1da3bcba2a1e8909efbc741e3d4c74c5d4f4/miner/block.go#L142-L147

https://github.com/celo-org/celo-blockchain/blob/c4ea1da3bcba2a1e8909efbc741e3d4c74c5d4f4/miner/block.go#L142-L147